.github/workflows/validate-crd.yaml

```diff
@@ -0,0 +1,45 @@
+name: Check CRD Version Update
+on:
+  pull_request:
+    paths:
+      - 'pkg/k8s/apis/cilium.io/client/crds/v1alpha1/*.yaml'
+
+jobs:
+  check-version:
+    runs-on: ubuntu-22.04
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: ${{ github.event_name == 'pull_request' && 2 || 0 }}
```

Comment:
This workflow is configured to run only on pull requests, so why check here if `github.event_name == 'pull_request'`?

Reply:
just checked . If we remove this It wun't detect the changes

Reply:
That's not an adequate reason to keep it though :)